### PR TITLE
Ui backend mapping

### PR DESCRIPTION
**Pull request by profileEnumerable**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,7 @@
 
 .vs/
 .idea/
+
+UI/Campus-UI/.vscode/
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply .gitignore:12]: ✔

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,7 @@
...
+
+UI/Campus-UI/debug.log
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply .gitignore:14]: ✔

File: UI/Campus-UI/src/app/data/dao/impl/CategoryDAOArray.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+import { ProjectModel } from '../../../model/Project';
 import { Observable, of } from 'rxjs';
 import { TestData } from '../../testData';
 
 export class CategoryDAOArray implements CategoryDAO {
 
-  add(category: Category): Observable<Category> {
+  add(category: ProjectModel): Observable<ProjectModel> {
 
     // если id пустой - генерируем его
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply UI/Campus-UI/src/app/data/dao/impl/CategoryDAOArray.ts]: ✔

**Review by antonAce — COMMENTED** [review]
Good!

File: UI/Campus-UI/src/app/data/dao/impl/CategoryDAOArray.ts [hunk 4]
```diff
@@ -21,26 +21,26 @@ export class CategoryDAOArray implements CategoryDAO {
...
-      cat => cat.title.toUpperCase().includes(title.toUpperCase()))
-      .sort((c1, c2) => c1.title.localeCompare(c2.title)));
+      cat => cat.name.toUpperCase().includes(title.toUpperCase()))
+      .sort((c1, c2) => c1.name.localeCompare(c2.name)));
   }
 
-  update(category: Category): Observable<Category> {
+  update(category: ProjectModel): Observable<ProjectModel> {
 
     const tmpCategory = TestData.categories.find(t => t.id === category.id); // обновляем по id
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply UI/Campus-UI/src/app/data/dao/impl/CategoryDAOArray.ts]: ✔

File: UI/Campus-UI/src/app/data/dao/interface/TaskDAO.ts [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+import { ProjectModel } from '../../../model/Project';
 import { Observable } from 'rxjs';
 import { Priority } from '../../../model/priority';
 
 export interface TaskDAO extends CommonDAO<Task> {
 
-  search(category: Category, searchText: string, status: boolean, priority: Priority): Observable<Task[]>;
+  search(category: ProjectModel, searchText: string, status: boolean, priority: Priority): Observable<Task[]>;
 
   // кол-во завершенных задач в заданной категории (если category === null, то для всех категорий)
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply UI/Campus-UI/src/app/data/dao/interface/TaskDAO.ts]: ✔

File: UI/Campus-UI/src/app/shared/services/project.service.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProjectService {
+
+  constructor(private http: HttpClient) { }
+
+  getAllUserProjects(): Observable<ProjectModel[]> {
+    const uri: string = environment.apiRoot + environment.projects + `?page=${1}&items=${20}`;
```
[the remark this answers is not in the document]
Comment (profileEnumerable) [reply UI/Campus-UI/src/app/shared/services/project.service.ts:16]: ✔

**Review by antonAce — APPROVED** [review]
At least it's better than before :1st_place_medal:

**Pull request merged**